src/Orchard.Cms.Web/Modules/Orchard.Setup/Services/SetupContext.cs

```diff
@@ -12,6 +12,7 @@ public class SetupContext
         public string DatabaseProvider { get; set; }
         public string DatabaseConnectionString { get; set; }
         public string DatabaseTablePrefix { get; set; }
+        public IDictionary<string, string> RecipeExtraParams { get; set; }
```

Comment:
Well, in the case anyone want to provide its own custom controller you are right, the same could be done with an anonymous.
But this morning I was thinking we can change current setup controller to act in a generic way passing all the key/values available in form posted to RecipeExtraParams (except those named as the properties in SetupContext).
Doing that in many cases we won't need to have our own controller. So, to have a custom setup we would need only a new view. That cannoy be done with an anonymous type as far as I know.
Next step would be we could add a custom shape to our theme that would replace custom setup form, then in many cases we won't need an extra module to have a custom setup.